### DISCUSSION TITLE
Guide for contributing new aws resource type support

### DISCUSSION THREAD

**Discussion by dehamzah** [cloud-custodian/cloud-custodian#9144] [AWS] [answered]
Hi, I have a use case to tag aws resource type that is currently not supported in c7n, is there a guide to help me ramp up so I can contribute? thank you in advance

**Comment by ajkerrigan** [comment, marked as answer]
Hi @dehamzah , thanks for your interest :wave: ! There are a couple sections of the documentation that should be helpful:

- [Adding New AWS Resources](https://cloudcustodian.io/docs/aws/contribute.html#adding-new-aws-resources)
- [Contributing to Cloud Custodian](https://cloudcustodian.io/docs/contribute.html)

It's probably also worth opening an issue for the resource you plan to add, so other folks will know there's work in progress if they have the same need.

Since real examples can be a helpful reference, you may get some value out of a GitHub search like [this](https://github.com/cloud-custodian/cloud-custodian/pulls?q=is%3Apr+is%3Amerged+in%3Atitle+add+aws+resource+) that shows some previous merged PRs that added AWS resources.

And of course if you have any questions or problems along the way, feel free to raise them in a draft PR or hop over to the `#development` channel [in Slack](https://cloudcustodian.io/community/).

**Reply by dehamzah** [reply]
Thank you @ajkerrigan , these are helpful, previously I only knew this https://cloudcustodian.io/docs/contribute.html and try to search the example merged PR. 

I have made my first PR here https://github.com/cloud-custodian/cloud-custodian/pull/9147
